src/test/java/uk/gov/justice/probation/courtcaseservice/controller/CourtCaseControllerTest.java

```diff
@@ -136,6 +136,24 @@ void getCourtCaseByCaseIdAndDefendantId_shouldReturnCourtCaseResponseNoCaseNo()
         assertThat(courtCase.getNumberOfPossibleMatches()).isEqualTo(2);
         assertThat(courtCase.getSession()).isSameAs(session);
 
+        verify(courtCaseService).getHearingByCaseIdAndDefendantId(CASE_ID, DEFENDANT_ID);
+        verify(offenderMatchService).getMatchCountByCaseIdAndDefendant(CASE_ID, DEFENDANT_ID);
+        verifyNoMoreInteractions(courtCaseService, offenderMatchService);
+    }
+
+    @Test
+    void getCourtCaseByHearingIdAndDefendantId_shouldReturnCourtCaseResponseNoCaseNo() {
+
+        when(offenderMatchService.getMatchCountByCaseIdAndDefendant(CASE_ID, DEFENDANT_ID)).thenReturn(Optional.of(2));
+        when(courtCaseService.getHearingByHearingIdAndDefendantId(CASE_ID, DEFENDANT_ID)).thenReturn(hearingEntity);
```

Comment:
Might be more apt to rename `CASE_ID` to `HEARING_ID` or break it out as a separate constant